tests_integration/testing_utils/assertions/test_result_assertions.py

```diff
@@ -5,8 +5,10 @@
 from tests_integration.test_utils import contains_row_with
 
 
-def assert_that_result_is_successful(result: CommandResult) -> None:
+def assert_that_result_is_successful(result: CommandResult, caplog=None) -> None:
     assert result.exit_code == 0, result.output
+    if caplog:
+        _assert_that_no_error_logs(caplog)
```

Comment:
IMO it should not be a part of `assert_that_result_is_successful`. Presence of error logs does not mean that result is automatically a failure.

For example running `snow sql -q "select 1"` using CLI with broken external plugin will still finish successfully.

IMO `assert_that_no_error_logs` should be a separate assertion.